### PR TITLE
POSHI-182 | master

### PR DESCRIPTION
**Pull request by Tim-Cao**
https://issues.liferay.com/browse/POSHI-182

@kenjiheigel

### CODE REVIEW

**Comment by Tim-Cao** [comment]
ci:test:acceptance-functional:nocompile

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
ci:test:acceptance-functional

**Comment by Tim-Cao** [comment]
Hi @kenjiheigel, according to ci results, the changes are safe.

**Comment by Tim-Cao** [comment]
FYI @kenjiheigel, I've resolved conflict

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
ci:test:acceptance-functional

**Comment by Tim-Cao** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
ci:test:acceptance-functional:nocompile

**Comment by Tim-Cao** [comment]
ci:test:acceptance-functional:nocompile

**Comment by kenjiheigel** [comment]
@Tim-Cao, sorry just getting to this.

File: modules/test/poshi/poshi-runner/src/main/java/com/liferay/poshi/runner/util/JSONUtil.java [hunk 1]
```diff
@@ -109,6 +109,25 @@ public static String getWithJSONPath(
...
+
+	public static void put(
+		JSONObject jsonObject, String key, List<String> value) {
+
+		jsonObject.put(key, value);
+	}
+
+	public static void put(JSONObject jsonObject, String key, String value) {
+		if (value.startsWith("{")) {
+			jsonObject.put(key, toJSONObject(value));
```
Comment (kenjiheigel) [inline modules/test/poshi/poshi-runner/src/main/java/com/liferay/poshi/runner/util/JSONUtil.java:124]: @Tim-Cao , why do you have 2 methods for this? In some cases you want to store a list in a JSONObject?

**Review by kenjiheigel — COMMENTED** [review]
@Tim-Cao, overall, this is a pretty big change. You're getting into storing data structures within Poshi variables, and we haven't planned out what that looks like long term. Lists are ok because that's a basic structure. JSON, although it's used a lot, is something we might want to think about more. I see the need for this, but I want to hear feedback from others before we do anything with this.

File: modules/test/poshi/poshi-runner/src/main/java/com/liferay/poshi/runner/util/JSONUtil.java (hunk 1 above)
Comment (Tim-Cao) [reply modules/test/poshi/poshi-runner/src/main/java/com/liferay/poshi/runner/util/JSONUtil.java:124]: Yep, the attribute value could a list, string, JSONObject, date type

**Comment by Tim-Cao** [comment]
ci:reopen

**Pull request closed without merging**